sdk/mixedreality/mixedreality-authentication/src/tracing.ts

```diff
@@ -1,59 +1,15 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { getTracer } from "@azure/core-tracing";
-import { Span, SpanOptions, SpanKind } from "@opentelemetry/api";
-import { OperationOptions } from "@azure/core-http";
-
-type OperationTracingOptions = OperationOptions["tracingOptions"];
+import { createSpanFunction } from "@azure/core-tracing";
 
 /**
  * Creates a span using the global tracer.
  * @internal
- * @param name The name of the operation being performed.
- * @param tracingOptions The options for the underlying http request.
  */
-export function createSpan<T extends OperationOptions>(
-  operationName: string,
-  operationOptions: T
-): { span: Span; updatedOptions: T } {
-  const tracer = getTracer();
-  const tracingOptions = operationOptions.tracingOptions || {};
-  const spanOptions: SpanOptions = {
-    ...tracingOptions.spanOptions,
-    kind: SpanKind.INTERNAL
-  };
-
-  const span = tracer.startSpan(`Azure.MixedReality.${operationName}`, spanOptions);
-
-  // Resource providers are documented here:
-  // https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers
-  span.setAttribute("Microsoft.MixedReality", "Microsoft.MixedReality");
-
-  let newSpanOptions = tracingOptions.spanOptions || {};
-  if (span.isRecording()) {
-    newSpanOptions = {
-      ...tracingOptions.spanOptions,
-      parent: span.context(),
-      attributes: {
-        ...spanOptions.attributes,
-        "Microsoft.MixedReality": "Microsoft.MixedReality"
-      }
-    };
-  }
-
-  const newTracingOptions: OperationTracingOptions = {
-    ...tracingOptions,
-    spanOptions: newSpanOptions
-  };
-
-  const newOperationOptions: T = {
-    ...operationOptions,
-    tracingOptions: newTracingOptions
-  };
-
-  return {
-    span,
-    updatedOptions: newOperationOptions
-  };
-}
+export const createSpan = createSpanFunction({
+  packagePrefix: "Azure.MixedReality",
+  // TODO: oddly enough the createSpan here was actually the attribute named
```

Comment:
```suggestion
```

@craigktreasure  - The old version of this was actually setting an _attribute_ named `Microsoft.MixedReality` to the value `Microsoft.MixedReality`. I _think_ you meant to set 'az.namespace' (if so, this change I did will fix it for you and we can just remove the TODO comment).

If not, let me know and I can revert it, but we'll have to discuss this a bit more when we move to the next version of opentelemetry.

Reply:
It's very likely that I made a mistake. You're probably right.

Reply:
Free fix then! :)